coding-articles/yehia-article-thegraph/Article.md

```diff
@@ -0,0 +1,660 @@
+
+```
+// .... Some Code .....
+// Inside component
+useEffect(()=>{
+    const fetchQuery = async()=>{
+       const result = await apolloClient.query({
+          query: pairSymbolsQuery
+       })
+       console.log(result.data)
+     }
+  fetchQuery()
+},[])
+```
+
+Now run your project `yarn dev` and check the console. You should see the same query result from the graph studio
+
+<figure>
+  <img src="./images/console_output.png" alt="uniswap deployed on the graph studio">
+  <figcaption align="center">Your project console</figcaption>
+</figure>
+
+Apollo clients have multiple hooks to [query](https://www.apollographql.com/docs/react/data/queries), [fetch](https://www.apollographql.com/docs/react/data/queries), and [cache](https://www.apollographql.com/docs/react/data/queries). Feel free to play around with them.
+Apollo client also support variables inside queries for example
+
+```
+await apolloClient.query({
+          query: pairSymbolsQuery
+          variables: { totalSupply: "1000" },
+       })
+
+```
+
+To identify a variable inside a query you need only to start it with `$` like `$totalSupply` variable
+
+```
+   query getTotalSupplyGt($totalSupply: BigDecimal) {
+    pairs(where:{totalSupply_gt: $totalSupply}) {
+        id
+        totalSupply
+    }
+   }
+```
+
+You can connect your frontend with any subgraph on the graph studio. Just copy the API URL. In the next section, we will create our subgraph to get the token names of every initialized pair to swap.
+
+## Build Your Subgraph To Catch All Pairs Of Token Names Initialized For Swap
+
+Before start implementing, The subgraph is defined by mainly three components
+
+- Schema
+- mapper
+- YAML file
+
+### Schema
+
+`schema.graphql` The schema file should define the data that you want to fetch. For example, if you're fetching token data, the token entity should look like this.
+
+```
+type Token @entity {
+  id: ID! # token address
+  name: String!
+  symbol: string!
+  decimals: BigInt!
+   ...
+}
+```
+
+### mapper
+
+Inside `src/` or `mapping/` you add the handler function. Handler functions are functions that trigger once an event has been called example
+
+```
+// PairCreated is an event inside the smart contract
+export function handlePairCreated(event: PairCreated): void {
+    // What will happen after the event is triggered
+}
+```
+
+### YAML file
+
+`subgraph.yaml` map handlers to the event. For example, to make the above handler work, we will link it with the event inside subgraph.yml this way
+
+```
+eventHandlers:
+        - event: PairCreated(indexed address,indexed address,address,uint256)
+          handler: handlePairCreated
+  file: ./<mapper file location>/map.ts
+```
+
+Now you understand the three primary files to create a subgraph; you can start by installing the graph-cli
+
+```
+yarn global add @graphprotocol/graph-cli
+```
+
+Create a directory and initialize the project
+
+```
+# Inside the project directory
```

Comment:
We should tell people to use the `graph init` command in a terminal at the same level as the next-app. That should be more clear.

Reply:
They can put it in any directory.  It is just my preference🌚  
I added this line. 
```
Create a directory and initialize the project outside the frontend folder
```
wdyt?